### PR TITLE
cpu/nrf52: add i2c driver and configs

### PR DESCRIPTION
**Pull request by PeterKietzmann**
### Contribution description

This PR adds a peripheral I2C driver for nrf52 MCUs.

### Issues/PRs references
Continuation of #7557

### CODE REVIEW

**Comment by PeterKietzmann** [comment]
@dylad @lebrush would you have a look at this PR?

File: cpu/nrf52/periph/i2c.c [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+ * @brief   If any of the 4 lower bits are set, the speed value is invalid
+ */
+#define INVALID_SPEED_MASK  (0x0f)
+
+/**
+ * @brief   Initialized bus locks (we have a maximum of two devices...)
+ */
+static mutex_t locks[] = {
+    MUTEX_INIT,
+    MUTEX_INIT
```
Comment (dylad) [inline cpu/nrf52/periph/i2c.c]: Should we initialize all bus locks even if only one is enabled through the `periph_conf.h` file ?

**Comment by PeterKietzmann** [comment]
@dylad thanks for your review. Please have a look at the latest commits. I've tested this implementation with a srf02 ultrasonic (I2C) sensor and the nrf52dk and the aconno board (acd52832)

File: cpu/nrf52/periph/i2c.c [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+    NRF_P0->PIN_CNF[i2c_config[bus].pin_scl] = (GPIO_PIN_CNF_DRIVE_S0D1 << GPIO_PIN_CNF_DRIVE_Pos);
+    dev(bus)->PSEL.SCL = i2c_config[bus].pin_scl;
+    dev(bus)->PSEL.SDA = i2c_config[bus].pin_sda;
+
+    /* shortcuts configuration */
+    dev(bus)->SHORTS = (TWIM_SHORTS_LASTTX_STOP_Enabled << TWIM_SHORTS_LASTTX_STOP_Pos) |
+                       (TWIM_SHORTS_LASTRX_STOP_Enabled << TWIM_SHORTS_LASTRX_STOP_Pos);
+
+    /* bus clock speed configuration */
+    dev(bus)->FREQUENCY = (TWIM_FREQUENCY_FREQUENCY_K100 << TWIM_FREQUENCY_FREQUENCY_Pos);
```
Comment (dylad) [inline cpu/nrf52/periph/i2c.c]: Please change `TWIM_FREQUENCY_FREQUENCY_K100` to `speed` so we can run I2C at 400 kHz.

Comment (PeterKietzmann) [reply cpu/nrf52/periph/i2c.c]: Good catch! I completely overlooked this one. Fixed in latest commit

**Comment by PeterKietzmann** [comment]
Ok to squash?

**Comment by PeterKietzmann** [comment]
Squashed. Sure we can wait if you want. But basically this PR is the same as #7557 which was open for about four months.

**Comment by dylad** [comment]
Let's merge this one :)

**Pull request merged**